exec/java-exec/src/main/java/org/apache/drill/exec/server/rest/QueryWrapper.java

```diff
@@ -46,15 +46,21 @@
   private static final double HEAP_MEMORY_FAILURE_THRESHOLD = 0.85;
 
   private final String query;
-
   private final String queryType;
+  private final Integer autoLimitRowCount;
 
   private static MemoryMXBean memMXBean = ManagementFactory.getMemoryMXBean();
 
   @JsonCreator
-  public QueryWrapper(@JsonProperty("query") String query, @JsonProperty("queryType") String queryType) {
+  public QueryWrapper(@JsonProperty("query") String query, @JsonProperty("queryType") String queryType, @JsonProperty("autoLimit") String autoLimit) {
     this.query = query;
     this.queryType = queryType.toUpperCase();
+    this.autoLimitRowCount = autoLimit.matches("[0-9]+") ? new Integer(autoLimit) : null;
+  }
+
+  @JsonCreator
+  public QueryWrapper(@JsonProperty("query") String query, @JsonProperty("queryType") String queryType) {
+    this(query, queryType, "");
```

Comment:
to avoid redundant pattern check : 
```suggestion
    this(query, queryType, null);
```

Reply:
Done in the latest set of commits